relay-server/src/actors/envelopes.rs

```diff
@@ -1805,8 +1822,10 @@ impl EnvelopeProcessor {
                 .map_err(ProcessingError::SerializeFailed)?
         });
 
-        let event_type = state.event_type().unwrap_or_default();
-        let mut event_item = Item::new(ItemType::from_event_type(event_type));
+        let mut event_item = match state.event_type().unwrap_or_default() {
+            EventType::Transaction => state.transaction_item.take().unwrap(),
```

Comment:
There's usually a better way than `unwrap()`. E.g. store `event_item` on the state instead of `event_type` and `transaction_item`, and then get the type from the item.